signer/keys/keys.go

```diff
@@ -16,18 +20,53 @@ var (
 	ErrFailedKeyGeneration = errors.New("notary-signer: failed to generate new key")
 )
 
-// Key represents all the information of a key, including the private and public bits
-type Key struct {
-	ID        string
-	Algorithm string
-	Public    [ed25519.PublicKeySize]byte
-	Private   *[ed25519.PrivateKeySize]byte
-}
-
 // HSMRSAKey represents the information for an HSMRSAKey with ObjectHandle for private portion
 type HSMRSAKey struct {
-	ID        string
-	Algorithm string
-	Public    []byte
-	Private   pkcs11.ObjectHandle
+	id      string
+	public  []byte
+	private pkcs11.ObjectHandle
+}
+
+// NewHSMRSAKey returns a HSMRSAKey
+func NewHSMRSAKey(public []byte, private pkcs11.ObjectHandle) *HSMRSAKey {
+	return &HSMRSAKey{
+		public:  public,
+		private: private,
+	}
+}
+
+// Algorithm implements a method of the data.Key interface
+func (rsa *HSMRSAKey) Algorithm() data.KeyAlgorithm {
+	return data.RSAKey
+}
+
+// ID implements a method of the data.Key interface
+func (rsa *HSMRSAKey) ID() string {
+	if rsa.id == "" {
+		pubK := data.NewTUFKey(rsa.Algorithm(), rsa.Public(), nil)
+		data, err := cjson.Marshal(&pubK)
```

Comment:
I copied TUFKey's ID method without thinking too hard about it. That is indeed a better idea.